scripts/init.sh

```diff
@@ -20,3 +20,9 @@ cargo install cargo-fuzz
 
 echo "Installing web packages"
 (cd web && pnpm install)
+
+echo "Download geth to replace the nix version"
+geth_package=geth-$(uname | tr '[:upper:]' '[:lower:]')-amd64-1.13.10-bc0be1b1
```

Comment:
I think its simpler to hardcode the source filenames based on the output of:

```shell
nix eval --impure --raw --expr 'builtins.currentSystem'
```

We need to support macos too, not just linux/amd64

Reply:
It seems the [download](https://geth.ethereum.org/downloads) page only provides the amd64 versions, and I tested the `darwin-amd64` did work in my M1.

Reply:
I think you need to look that download page a bit more - there are links to downloads for many architectures:

<img width="1202" alt="Screenshot 2024-01-18 at 15 20 55" src="https://github.com/Snowfork/snowbridge/assets/117534/4131c94d-9a82-4d6f-9f3d-ea235d5bc45b">

So depending on the value of `builtins.currentSystem`, which should be one of the architectures listed in `flake.nix`, download the appropriate binary.

Reply:
https://github.com/Snowfork/snowbridge/pull/981/commits/fb5366f196f78b7f950763a5fbc087385d1dcb62

Btw: `nix eval --impure --raw --expr 'builtins.currentSystem'` return `aarch64-darwin` in my laptop which does not match the `arm64-darwin` in the download link so switch to `uname -m` instead.